### PR TITLE
feat: add test for function build cancelation

### PR DESCRIPTION
**Pull request by guilhem-barthes**
Fixes FL-1659

## Companion PR

- https://github.com/Substra/substra-backend/pull/997

### CODE REVIEW

File: tests/test_docker_image_build.py [hunk 1]
```diff
@@ -86,3 +86,39 @@ def test_function_build_order(factory, cfg, client, worker):
...
+    dockerfile_test = get_dockerfile(
+        docker_image, function_category, extra_instructions="ENV test=test_function_cancelled_cp_test"
+    )
+
+    function_wait_spec = factory.create_function(function_category, dockerfile=dockerfile_wait)
+    function_test_spec = factory.create_function(function_category, dockerfile=dockerfile_test)
+
+    inputs = default_dataset.opener_input + default_dataset.train_data_sample_inputs
+    cp = factory.create_compute_plan()
+    function = client.add_function(function_wait_spec)
```
Comment (SdgJlbl) [inline tests/test_docker_image_build.py:113]: WDYT of not calling both functions by the same variable name?

File: tests/test_docker_image_build.py [hunk 2]
```diff
@@ -86,3 +86,39 @@ def test_function_build_order(factory, cfg, client, worker):
...
+    function = client.add_function(function_test_spec)
+    cp.create_traintask(function=function, worker=worker, inputs=inputs)
+    client.add_compute_plan(cp)
+
+    client.cancel_compute_plan(cp.key)
+    cp = client.wait_compute_plan(cp.key, raise_on_failure=False, timeout=cfg.options.future_timeout)
+    assert cp.status == "PLAN_STATUS_CANCELED"
+
+    function = client.wait_function(function.key, raise_on_failure=False, timeout=cfg.options.future_timeout)
+    assert function.status == "FUNCTION_STATUS_CANCELED"
```
Comment (SdgJlbl) [inline tests/test_docker_image_build.py:124]: We are not checking anything on the first function added 🤔

**Review by SdgJlbl — COMMENTED** [review]
Thanks for your work, I think we are missing some status check on one of the two functions though.

File: tests/test_docker_image_build.py (hunk 2 above)
Comment (guilhem-barthes) [reply tests/test_docker_image_build.py:124]: What do you think would make sense to check in this case?

File: tests/test_docker_image_build.py (hunk 1 above)
Comment (guilhem-barthes) [reply tests/test_docker_image_build.py:113]: No strong opinion

File: tests/test_docker_image_build.py (hunk 2 above)
Comment (SdgJlbl) [reply tests/test_docker_image_build.py:124]: That it's not failed or canceled, I guess?

**Review by SdgJlbl — APPROVED** [review]

**Pull request merged**